platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1025,35 +700,53 @@ namespace platoon_strategic
         }
     }
 
-    void PlatoonStrategicPlugin::mob_op_cb_leader(const cav_msgs::MobilityOperation& msg)
+    // UCLA: add condition to account for frontal joining 
+    void PlatoonStrategicIHPPlugin::mob_op_cb_leader(const cav_msgs::MobilityOperation& msg)
     {   
         std::string strategyParams = msg.strategy_params;
-        std::string senderId = msg.header.sender_id;
-        std::string platoonId = msg.header.plan_id;
+        std::string senderId = msg.header.sender_id; //sender is target platoon leader
```

Comment:
Somewhere in here it would be good to note that if host is a platoon leader it is ignoring incoming INFO messages from any candidate joiner.  That is definitely not obvious, and it is not obvious that it is intentional.